Split channels_last test cases for tensor conversion OpInfos

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#67368 Split channels_last test cases for tensor conversion OpInfos**

This PR adds an addition test variant for the tensor conversion
functions (bfloat16, char, long, ...) that tests channels_last. This is
because some backends (mostly just functorch right now) don't have
channels last handling and may want to test that separately from the
more general case of these operations.

Test Plan:
- wait for tests

Differential Revision: [D31972959](https://our.internmc.facebook.com/intern/diff/D31972959)

zou3519 committed:
Split channels_last test cases for tensor conversion OpInfos

This PR adds an addition test variant for the tensor conversion
functions (bfloat16, char, long, ...) that tests channels_last. This is
because some backends (mostly just functorch right now) don't have
channels last handling and may want to test that separately from the
more general case of these operations.

Test Plan:
- wait for tests

[ghstack-poisoned]

mruberry reviewed (approved):
Cool refinement; just give the tests an opportunity to run

zou3519 committed:
Update on "Split channels_last test cases for tensor conversion OpInfos"


This PR adds an addition test variant for the tensor conversion
functions (bfloat16, char, long, ...) that tests channels_last. This is
because some backends (mostly just functorch right now) don't have
channels last handling and may want to test that separately from the
more general case of these operations.

Test Plan:
- wait for tests

Differential Revision: [D31972959](https://our.internmc.facebook.com/intern/diff/D31972959)

[ghstack-poisoned]
